back/app/models/phase.rb

```diff
@@ -264,6 +240,20 @@ def started?
     start_at <= Time.zone.now
   end
 
+  # Used for validations (which are hard to delegate through the participation method)
+  def voting?
+    participation_method == 'voting'
+  end
+
+  # Used for validations (which are hard to delegate through the participation method)
+  def native_survey?
+    participation_method == 'native_survey'
+  end
+
+  def method
+    @method ||= Factory.instance.participation_method_for(self)
+  end
```

Comment:
It's probably a good idea to give this a better name, because `method` is also a built-in in Ruby method. If we overwrite it, we risk to break external behavior. I suggest `participation_method`.

Reply:
Or `pmethod` if that's too long

Reply:
Sounds good! I propose to use `participation_method`, replacing the string version. It's hard to estimate the effort, but it feels doable to me (I just wonder if we would introduce bugs with inputs that are not yet persisted). I'll look into this today.

Reply:
@kogre @IvaKop I've been spending my time until now making this change and it's getting out of hand: https://github.com/CitizenLabDotCo/citizenlab/pull/8477/files. I needed to introduce a `raw_participation_method` to make the validations work and because `participation_method` is still used in different places throughout the codebase. Also, a lot of specs fail after this change, some of which are really hard to fix (e.g. voting_counts suddenly being wrong) https://app.circleci.com/pipelines/github/CitizenLabDotCo/citizenlab?branch=TAN-2342-replace-participation-method

So I'm not sure if we should spend much more time on this. Instead, let's go for the simpler solution by keeping the string valued `participation_method` and renaming `method` to `pmethod` (and call that everywhere in the code base)? The more difficult solution of replacing `participation_method` is then represented by this ticket: https://www.notion.so/govocal/Replace-phase-participation_method-7f081a8dffdd4a3eb5b6b89c82142bb8?pvs=4

Should we remove the Factory method (in the PR changes it's not doing much anymore, except dealing with a project as argument -- I think we could move that code somewhere else, though not sure where exactly)?

Reply:
Hey @sebastienhoorens , totally agree that this isn't worth that much time or difficulty, just going with `pmethod` (or not `method`) for the method name is enough.